lib/checks/001-deprecations.js

```diff
@@ -21,32 +21,32 @@ checkDeprecations = function checkDeprecations(theme, themePath) {
                 ruleCode: 'GS001-DEPR-IMG'
             },
             {
-                helperRegEx: '{{author.image}}',
+                helperRegEx: new RegExp(/{{author\.image}}/),
```

Comment:
Escaping the dot will fix the issue :)

Consistency Notes:

- the use of `new RegExp()` isn't really needed, it's useful if you need to build a dynamic regex, e.g. you want to concatenate a variable in like:
`var myRegex = new Regex(myVar1+ myVar1)`
However here I don't think we ever really need it as regexes can be defined as `helperRegex: /{{post\.image}}/`
- Some regexes have the `g` flag, some don't - I think this should be consistent. I believe the `g` flag will make the rule ind ALL instances of the match, and without it will only match the first.
- If you are using `new RegExp()` flags should be passed as the second argument, E.g. `new RegExp(/{{post\.image}}/, 'g')` otherwise it would be `helperRegex: /{{post\.image}}/g`
- `{` is also a special char in regexes, I was going to suggest escaping the first one only to be clear this is not meant to be a special char but testing with https://regex101.com/ shows this is really not necessary.

TLDR; this will fix the bug, there are also some inconsistencies.